### ISSUE TITLE
question for "evaluate.py"

### ISSUE THREAD

**Issue by amiltonwong** [DrSleep/tensorflow-deeplab-resnet#21]
Hi, @DrSleep 

I had run the evaluation:
$  python evaluate.py --data-dir /data/VOC_dataset/voc2012_trainval/JPEGImages --data-list ./dataset/val.txt 

However, using the provided model: deeplab_resnet.ckpt, the Mean IoU score: 0.047 is very low.
Could you tell me what's wrong with that?  THX!

Here is the output:
```
(python_2.7_tf_0.12) root@milton-OptiPlex-9010:/data/code/tensorflow-deeplab-resnet# python evaluate.py --data-dir /data/VOC_dataset/voc2012_trainval/JPEGImages --data-list ./dataset/val.txt 
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcudnn.so locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcurand.so locally
I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
I tensorflow/core/common_runtime/gpu/gpu_device.cc:885] Found device 0 with properties: 
name: GeForce GTX TITAN
major: 3 minor: 5 memoryClockRate (GHz) 0.8755
pciBusID 0000:01:00.0
Total memory: 5.94GiB
Free memory: 5.56GiB
I tensorflow/core/common_runtime/gpu/gpu_device.cc:906] DMA: 0 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:916] 0:   Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:975] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX TITAN, pci bus id: 0000:01:00.0)
Restored model parameters from ./deeplab_resnet.ckpt
step 0
step 100
step 200
step 300
step 400
step 500
step 600
step 700
step 800
step 900
step 1000
step 1100
step 1200
step 1300
step 1400
Mean IoU: 0.047
(python_2.7_tf_0.12) root@milton-OptiPlex-9010:/data/code/tensorflow-deeplab-resnet# 

```

**Comment by DrSleep** [comment]
Hi, would you please run the inference.py script on some image and share the result?

**Comment by amiltonwong** [comment]
Thanks @DrSleep ,
I figure it out: I should use [SegmentationClassAug](https://www.dropbox.com/s/oeu149j8qtbs1x0/SegmentationClassAug.zip?dl=0)

**Comment by amiltonwong** [comment]
From the [comparison](https://www.dropbox.com/s/wp240m8afvq2of5/compare.jpg?dl=0) on appearance between SegmentationClass and SegmentationClassAug, it seems the evaluation result for SegmentationClassAug is not equivalent to the original one SegmentationClass
Could you elaborate a bit about "SegmentationClassAug" ? THX~

**Comment by DrSleep** [comment]
Thank for letting me know. 

> Mean IoU: 0.869
Do you use the whole validation set? 

As for the difference between Class and ClassAug, the latter one does not contain the palette (the colour map). From the [DeepLab FAQ page](http://ccvl.stat.ucla.edu/deeplab_faq/):

> Q: The label read by IMAGE_SEG_DATA is not the same as expected. Do you pre-process the annotations?
> 
> A: You need to save the png annotations without the colormap; otherwise OpenCV will read the wrong values. See the script at matlab/my_script/SavePngAsRawPng.m for reference.

**Comment by amiltonwong** [comment]
@DrSleep ,
Thanks for your reply.
For this evaluation, I use the original [val.txt](https://gist.github.com/amiltonwong/d3d311d6389caedad8598e5c899ca7fa)

**Comment by DrSleep** [comment]
Thanks again for letting me know about the problem and providing the link to the dataset!

**Comment by sunbin1205** [comment]
@amiltonwong   
I used the SegmentationClassAug dataset, but the result of running evalut.py was Mean IoU: 0.017.
Why is that? And I also used deeplab_resnet_init. ckpt (because I didn't download the deeplab_resnet.ckpt) file to run the inference. py, and the result was all colored specks. Is the ckpt file the cause of the problem?
look forward to your reply!
